### PR TITLE
Add ViewGroup extension to allow concisely modifying of the layoutParams

### PR DESCRIPTION
**Pull request by steurt**
Added ViewGroup extension to allow concisely modifying of the layoutParams.

This allows following usage:
```
viewGroup.layoutParamsAs<ViewGroup.MarginLayoutParams> {
    updateMarginsRelative(start = marginStart)
}
```

Instead of:
```
val marginLayoutParams = viewGroup.layoutParams as ViewGroup.MarginLayoutParams
marginLayoutParams.updateMarginsRelative(start = 16)
viewGroup.layoutParams = marginLayoutParams
```

Was unable to update the `api` due to reified generics issue https://github.com/android/android-ktx/issues/235.

### CODE REVIEW

**Comment by steurt** [comment]
I signed it!

**Comment by steurt** [comment]
I signed it!

**Comment by steurt** [comment]
I signed it!

**Comment by steurt** [comment]
I signed it again!

File: src/main/java/androidx/view/ViewGroup.kt [hunk 1]
```diff
@@ -77,6 +77,21 @@ val ViewGroup.children: Sequence<View>
...
     }
 
+/**
+ * Executes [block] with a typed version of the ViewGroup's layoutParams and reassigns the
+ * layoutParams with the updated version.
+ *
+ * @see ViewGroup.getLayoutParams
+ * @see ViewGroup.setLayoutParams
+ **/
+inline fun <reified T> ViewGroup.layoutParamsAs(
```
Comment (romainguy) [inline src/main/java/androidx/view/ViewGroup.kt]: The name does not match the behavior. I would expect this method to simply be a get+cast, not to trigger a requestLayout via setting the LPs.

Comment (steurt) [reply src/main/java/androidx/view/ViewGroup.kt]: Do you think `withLayoutParams` would be better? Suggestions?

Comment (MGaetan89) [reply src/main/java/androidx/view/ViewGroup.kt]: `updateLayoutParams`?

Comment (steurt) [reply src/main/java/androidx/view/ViewGroup.kt]: @romainguy would `updateLayoutParams` or `updateLayoutParamsAs` work for you?

**Comment by steurt** [comment]
Renamed from `layoutParamsAs` to `updateLayoutParamsAs`. The usage then becomes:
```
viewGroup.updateLayoutParamsAs<ViewGroup.MarginLayoutParams> {
    updateMarginsRelative(start = marginStart)
}
```

Added non-generic variant `updateLayoutParams` for simply updating the layout params as `ViewGroup.LayoutParams`. This allows following usage:
```
viewGroup.updateLayoutParams { width = 500 }
```

**Comment by steurt** [comment]
Renamed from `layoutParamsAs` to `updateLayoutParamsAs`. The usage then becomes:
```
viewGroup.updateLayoutParamsAs<ViewGroup.MarginLayoutParams> {
    updateMarginsRelative(start = marginStart)
}
```

Added non-generic variant `updateLayoutParams` for simply updating the layout params as `ViewGroup.LayoutParams`. This allows following usage:
```
viewGroup.updateLayoutParams { width = 500 }
```

**Comment by steurt** [comment]
@romainguy any thoughts?

**Comment by steurt** [comment]
I can name the generic version `updateLayoutParams` as well. However, in doing so the non-generic and generic variant will have the same JVM signature, i.e. a platform declaration clash.

This can be solved by specifying a different `@JvmName` annotation for each.
Perhaps:
```
@JvmName("updateViewGroupLayoutParams")
inline fun ViewGroup.updateLayoutParams(block: ViewGroup.LayoutParams.() -> Unit)

@JvmName("updateLayoutParams")
inline fun <reified T> ViewGroup.updateLayoutParams(block: T.() -> Unit)
```

Will this do?

**Comment by romainguy** [comment]
Should we even have the non generic version at this point? @JakeWharton?

**Comment by JakeWharton** [comment]
Playing with it locally.

**Comment by JakeWharton** [comment]
How about this? https://github.com/android/android-ktx/compare/steurt-master

**Comment by steurt** [comment]
@JakeWharton seems like a good setup to me!

**Comment by romainguy** [comment]
LGTM

**Pull request closed without merging**